### ISSUE TITLE
Log filtering does not work as expected at logDNA agent level in Kubernetes environment

### ISSUE THREAD

**Issue by tapas-saha** [logdna/logdna-agent#98]
I used the following to exclude certain logs from being collected by the `logdna-agent` in `k8s`.
```
- name: LOGDNA_EXCLUDE
          value: /var/log/!(containers)*, /var/log/containers/*_kube-system_*, /var/log/containers/*_kube-public_*, /var/log/containers/*_default_*
```
With the above env setting, I see that the logs pertaining to `kube-system`, `kube-public` and `default` namespaces are not getting ingested as expected. However, the `/var/log/!(containers)*` doesn't seem to work well, i.e. I see multiple non-container logs getting pushed to the logDNA server. I tried with both `logdna/logdna-agent-v2:stable` and `logdna/logdna-agent:latest` and used the `.yaml` files as they are (except adding `LOGDNA_EXCLUDE` env variable to the pod template.

Could anyone please take a look and help me understand what could be wrong here? Apologies, if it is more of a question rather than an issue as such.

**Comment by tapas-saha** [comment]
I figured out why my filtering was not working earlier. Looks like the paths are considered at its immediate level only and is not recursive w.r.t. the multiple levels of further directories inside. I added the following two paths to the exclusion list.

```
/var/log/!(containers)*/*
/var/log/!(containers)*/*/*
```

This helped getting the files under `/var/log/dir1` and `/var/log/dir2/dir3/` ignored by the agent.

However, I am still looking for an answer to see if the behavior I notice is correct by design.
